[WIP] DVC: Add new page and for several sub commands (#4225)
tldr-pages/tldr: pages/common/dvc-diff.md

# dvc diff

> > Show changes in DVC tracked file and directories added/deleted/modified.

Reviewer: Perhaps this will read better?
```suggestion
> Show changes in DVC tracked file and directories.
```

> More information: <https://dvc.org/doc/command-reference/diff>.

> - Compare DVC tracked file from different Git commit/tag/branch w.r.t current workspace:

Reviewer: Perhaps this will read better?
```suggestion
- Compare DVC tracked files from different Git commits, tags, and branches w.r.t the current workspace:
```

`dvc diff {{commit_hash/tag/branch}}`

> - Compare DVC tracked file in Git b_rev w.r.t a_rev:

Reviewer: Perhaps this will read better?
```suggestion
- Compare the differences in DVC tracked files across 2 Git commits:
```

Author: The comparison is not commutative. i.e 
- `dvc diff b_rev a_rev` is not equal to `dvc diff a_rev b_rev`.

Thus deleted in one becomes added in other. I think we need to communicate this too.

Reviewer: Ah, I see - good point. What  about this then?
```suggestion
- Compare the changes in DVC tracked files from 1 git commit to another:
```

> `dvc diff b_rev a_rev`

Reviewer: Token syntax?
```suggestion
`dvc diff {{revision_b}} {{revision_a}}`
```

> - Compare DVC tracked file, along with their latest hash:

Reviewer: Perhaps this will read better?
```suggestion
- Compare DVC tracked files, along with their latest hash:
```

`dvc diff --show-hash {{commit}}`

> - Compare DVC tracked file, get output in JSON format:

Reviewer: Perhaps this will read better?
```suggestion
- Compare DVC tracked files, displaying the output as JSON:
```

`dvc diff --show-json --show-hash {{commit}}`

> - Compare DVC tracked file, get output in Markdown format:

Reviewer: Perhaps this will read better?
```suggestion
- Compare DVC tracked files, displaying the output as Markdown:
```
